dg.go

```diff
@@ -293,8 +293,10 @@ func (n *node[NodeType]) resolveNode(newStatus ResolutionStatus) error {
 		return ErrNodeDeleted{n.id}
 	}
 	if n.status != Waiting {
-		if n.status == Resolved || n.status == Unresolvable {
+		if n.status == Resolved {
 			return ErrNodeResolutionAlreadySet{n.id, n.status, newStatus}
+		} else if n.status == Unresolvable {
+			return nil // Allow nodes to be unresolved multiple times. But no processing is required.
 		} else {
 			return ErrNodeResolutionUnknown{n.id, n.status}
 		}
```

Comment:
@jaredoconnell, are you sure that this is what you want?

As a thought experiment, I think it would be good to consider this as a state machine and to list out all of the state-to-state transitions:

| From/To | Waiting | Resolved | Unresolvable |
| :- | :-: | :-: | :-: |
| **Waiting** | 🆗 | ✅ | ✅ |
| **Resolved** | ❌ | ❌ | ❌ |
| **Unresolvable** | ❌ | ❓ | 🆗 |

So, it's OK to transition from `Waiting` to `Waiting` (it's a no-op); and it is appropriate to transition from `Waiting` to either of the other two states.  It is an error to transition from `Resolved` to any other state, including `Resolved`.

For `Unresolvable`, things are more interesting.  Presumably it should be OK to transition to `Unresolvable` (this would be a no-op).  And, presumably it should be an error to transition back to `Waiting`.  The question is, should it be acceptable to transition from `Unresolvable` to `Resolved`...and I suspect the answer is no -- that indicates that the previous transition to `Unresolvable` was erroneous.

So, with that in hand, looking at the code, I think we're implementing the transitions from the `Waiting` state correctly.  Likewise, I think we have transitions from `Resolved` covered (in both the existing and your proposed code).  As for `Unresolvable`, I think what you are trying to change is to permit the no-op transition from `Unresolvable` to `Unresolvable`...but, your proposed change also removes the error detection from the other two transitions out of `Unresolvable`.

Is that really what you want?

Reply:
Yeah. We don't want to allow resolving after unresolution. Fixed.

Reply:
Your initial change would not have actually _allowed_ the transition from `Unresovled` to `Resolved`, it would have just failed silently....